server/main.py

```diff
@@ -36,11 +36,11 @@ async def upload(key: Optional[str] = Form(None), encoded_content: List[str] = F
         directory_key = f"./storage/{key_directory}"
         os.makedirs(directory_key , exist_ok=True)
         existing_files_count = len([name for name in os.listdir(directory_key) if name.startswith("encodedtxt")])
-        for i, encoded_item in enumerate(separated_strings, start=existing_files_count + 1):
+        for i, encoded_item in enumerate(encoded_content, start=existing_files_count + 1):
```

Comment:
This changes should only be present in update flow and not in upload. If i remember correctly in upload flow we are not allowing to update. If the key exists in store we just return " duplicate key - if you are trying to append please use append functionality"